### ISSUE TITLE
Use Swagger-Annotated Code As Input To swagger-codegen

### ISSUE THREAD

**Issue by packleader** [swagger-api/swagger-codegen#5694]
##### Description
**Feature Request: Enable swagger-codegen to generate a client API by inspecting a service's JAR/WAR for Swagger annotations.**
As a RESTful service developer, I want to be able to generate and publish client JARs for my service as part of that service's build process.  This will enable other developers to more easily integrate with my service.  To accomplish this, I need to run swagger-codegen before the service is deployed and swagger.json is generated.  I think the best approach is to have swagger-codegen inspect either my source files or compiled class files and use them as input.

##### Suggest a Fix
I have developed a Maven plugin that will scan a JAR file for Swagger annotations, generate swagger.json based on those annotations, then call swagger-codegen to generate a client API from the swagger.json.  I was planning to release the plugin as a standalone OSS project, but then I thought it might make sense to contribute it as a new feature to this project.

**Comment by wing328** [comment]
> I have developed a Maven plugin that will scan a JAR file for Swagger annotations, generate swagger.json based on those annotations

IMO the plug-in should be standalone as the auto-generated swagger spec can be consumed not only by Swagger Codegen but other tools as well such as ReadyAPI, Amazon API gateway, etc

**Comment by packleader** [comment]
I agree that a plugin who's only job is to generate the swagger spec from code would be best served as a separate project.  However, I was thinking of a more integrated solution - a single plugin to do all the work of parsing the source JAR file and generating the client API.

**Comment by ePaul** [comment]
I don't think this should be part of swagger-codegen (the core library just takes a swagger-core Swagger object, other parts (e.g. the maven plugin and CLI) add parsing of the JSON/YAML to it). Though I can imagine that there is a tool which does both steps (extract OpenAPI definition from Swagger annotations, and generate stuff from that) in one build.

**Comment by wing328** [comment]
> However, I was thinking of a more integrated solution - a single plugin to do all the work of parsing the source JAR file and generating the client API.

My suggestion is to start with "parsing the source JAR file" first and put down "generating the client API" as a day-2 requirement.
